tools/install_opte.sh

```diff
@@ -52,36 +53,44 @@ function sha_from_url {
     curl -L "$SHA_URL" 2> /dev/null | cut -d ' ' -f 1
 }
 
-OPTE_P5P_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G0MRWX9Y0X46HBEJBW245DJY/PM097Agvf89uKmVRZ890z6saoeLp6RCcVsbYRa5PDv9DnLDT/01G0MRX6GMBV34CNANABXZXX25/01G0MSFZZWPFEQBW7JRS7ST99G/opte-0.1.58.p5p"
-OPTE_P5P_SHA_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G0MRWX9Y0X46HBEJBW245DJY/PM097Agvf89uKmVRZ890z6saoeLp6RCcVsbYRa5PDv9DnLDT/01G0MRX6GMBV34CNANABXZXX25/01G0MSG01CGP6TH9THNY39G88Z/opte-0.1.58.p5p.sha256"
-OPTE_P5P_REPO_PATH="$OUT_DIR/$(basename "$OPTE_P5P_URL")"
-XDE_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G0DM53XR4E008D6ET5T8DXP6/wBWo0Jsg1AG19toIyAY23xAWhzmuNKmAsF6tL18ypZODNuHK/01G0DM5DMQHF5B89VGHZ05Z4E0/01G0DMHNYQ1NS7DBX8VG3JPAP0/xde"
-XDE_SHA_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G0DM53XR4E008D6ET5T8DXP6/wBWo0Jsg1AG19toIyAY23xAWhzmuNKmAsF6tL18ypZODNuHK/01G0DM5DMQHF5B89VGHZ05Z4E0/01G0DMHP47353961S3ETXBSD2T/xde.sha256"
+# Add a pkg repo by path. This will also run `pkg update`, handling the case
+# gracefully where there are no updates required
+function add_pkg_repo_and_update {
+    local REPO_PATH="$1"
+    pkg set-publisher -p "$REPO_PATH" --search-first
+    RC=0
+    pkg update || RC=$?;
+    if [[ "$RC" -eq 0 ]] || [[ "$RC" -eq 4 ]]; then
+        return 0
+    else
+        return "$RC"
+    fi
+}
 
-download_and_check_sha "$OPTE_P5P_URL" "$(sha_from_url "$OPTE_P5P_SHA_URL")"
-XDE_SHA="$(sha_from_url "$XDE_SHA_URL")"
-download_and_check_sha "$XDE_URL" "$XDE_SHA"
+# The `helios-netdev` provides the XDE kernel driver and the `opteadm` userland
+# tool for interacting with it.
+HELIOS_NETDEV_REPO_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G11AT7E4XV9J1J54GE2YDJT6/CB4WF4BVgnbvf5NI573z9osAV2LNIKogPtWJ5sfW2cNxUYQO/01G11ATFVTWAC2HSNV148PQ4ER/01G11B5MPQRBX3Q5EF45YDAW6Q/opte-0.1.60.p5p"
+HELIOS_NETDEV_REPO_SHA_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G11AT7E4XV9J1J54GE2YDJT6/CB4WF4BVgnbvf5NI573z9osAV2LNIKogPtWJ5sfW2cNxUYQO/01G11ATFVTWAC2HSNV148PQ4ER/01G11B5MR60H4N13NJKGWEEA69/opte-0.1.60.p5p.sha256"
+HELIOS_NETDEV_REPO_PATH="$XDE_DIR/$(basename "$HELIOS_NETDEV_REPO_URL")"
 
-# Move the XDE driver into it the expected location to allow operating on it
-# with `add_drv` and `rem_drv`
-DRIVER_DIR="/kernel/drv/amd64"
-XDE_FILENAME="$(basename "$XDE_URL")"
-XDE_PATH="$DRIVER_DIR/$XDE_FILENAME"
-if ! [[ -f "$XDE_PATH" ]] || [[ "$XDE_SHA" != "$(file_sha "$XDE_PATH")" ]]; then
-    echo "Replacing XDE driver"
-    mv -f "$OUT_DIR/$XDE_FILENAME" "$XDE_PATH"
-else
-    echo "XDE driver already exists with correct SHA"
-fi
+# The XDE repo provides a full OS/Net incorporation, with updated kernel bits
+# that the `xde` kernel module and OPTE rely on.
```

Comment:
FYI: It's a bug that this package is not also going by the publisher name `helios-netdev`.

- https://github.com/oxidecomputer/os-build/issues/2

Reply:
Great -- just let me know when that's merged and I'll update this script to point to the new release.